apps/sms/views/conversation/js/conversation.js

```diff
@@ -672,7 +676,7 @@ var ConversationView = {
     }
 
     // TODO move most of back() here: Bug 1010223
-    if (!this.isConversationPanel(Threads.currentId, nextPanel)) {
+    if (!this.isConversationPanel(this.activeThread, nextPanel)) {
```

Comment:
you need to pass the id here; so either `Threads.currentId` or `this.activeThread.id` (I prefer the second one because this is one dependency less to Threads)